pkg/config/schema/codegen/tools/collections.main.go

```diff
@@ -28,14 +29,20 @@ import (
 
 // Utility for generating collections.gen.go. Called from gen.go
 func main() {
-	if len(os.Args) != 4 {
+	if len(os.Args) != 4 && len(os.Args) != 7 {
 		fmt.Printf("Invalid args: %v", os.Args)
 		os.Exit(-1)
 	}
 
 	pkg := os.Args[1]
 	input := os.Args[2]
 	output := os.Args[3]
+	var splitOn, splitOutput, splitPrefix string
+	if len(os.Args) > 4 {
```

Comment:
Is the intent to panic if there are only 4 or 5 elements?

Reply:
I think L32 should ensure its either exactly 4 or 7. This code has gotten pretty sketchy over time though (especially with this PR)

Reply:
ah right ... missed that. Maybe expand the error message a bit to explain the valid conditions more clearly?